packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpMvsApi.ts

```diff
@@ -404,14 +405,28 @@ export class FtpMvsApi extends AbstractFtpApi implements MainframeInteraction.IM
             return loadResult.apiResponse.etag as string;
         }
 
-        const tmpFileName = tmp.tmpNameSync();
-        const options: zosfiles.IDownloadOptions = {
-            binary: false,
-            file: tmpFileName,
-        };
-        const loadResult = await this.getContents(dataSetName, options);
-        fs.rmSync(tmpFileName, { force: true });
-        return loadResult.apiResponse.etag as string;
+        // Create a temporary directory and unique filename
+        const tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), "zowe-ftp-mvs-"));
+        const tmpFileName = path.join(tmpDir, `temp-${crypto.randomUUID()}.dat`);
+
+        try {
+            const options: zosfiles.IDownloadOptions = {
+                binary: false,
+                file: tmpFileName,
+            };
+            const loadResult = await this.getContents(dataSetName, options);
+            return loadResult.apiResponse.etag as string;
+        } finally {
+            // Clean up temporary file and directory
+            try {
+                fs.rmSync(tmpFileName, { force: true });
+                fs.rmSync(tmpDir, { force: true, recursive: true });
```

Comment:
The combination of both function calls seems redundant, if our temp file is the only one inside the temp dir.

Could we either remove the 1st `rmSync` call, or do the following?
```suggestion
                fs.rmSync(tmpFileName, { force: true });
                fs.rmSync(tmpDir, { force: true });
```

Reply:
Resolved in 863c560